File: content/examples/other/leetcode/problems/invert-flip-binary-tree/example-1/solution.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def invert_tree_recursive(root):
    if root is None:
        return None
    left = invert_tree_recursive(root.left)
    right = invert_tree_recursive(root.right)
    root.left = right
    root.right = left
    return root


if __name__ == "__main__":
    root = TreeNode(4, TreeNode(2, TreeNode(1), TreeNode(3)), TreeNode(7, TreeNode(6), TreeNode(9)))
    print(invert_tree_recursive(root).left.val)

File: content/examples/other/leetcode/problems/invert-flip-binary-tree/example-1/Solution.java
public class Solution {
  static class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int val) {
      this.val = val;
    }

    TreeNode(int val, TreeNode left, TreeNode right) {
      this.val = val;
      this.left = left;
      this.right = right;
    }
  }

  static TreeNode invertTreeRecursive(TreeNode root) {
    if (root == null) return null;
    TreeNode left = invertTreeRecursive(root.left);
    TreeNode right = invertTreeRecursive(root.right);
    root.left = right;
    root.right = left;
    return root;
  }

  public static void main(String[] args) {
    TreeNode root =
        new TreeNode(
            4,
            new TreeNode(2, new TreeNode(1), new TreeNode(3)),
            new TreeNode(7, new TreeNode(6), new TreeNode(9)));
    System.out.println(invertTreeRecursive(root).left.val);
  }
}
